### PR TITLE
Fix events reshuffling

### PR DESCRIPTION
**Pull request by nebolax**
Removes the possibility of the appearance of duplicated indices when reshuffling events.

### CODE REVIEW

File: rotkehlchen/tests/unit/decoders/test_main.py [hunk 1]
```diff
@@ -489,3 +491,25 @@ def test_eth_deposit(
...
+    """
+    Tests that `maybe_reshuffle_events` works correctly.
+    Especially tests that there are no duplicated indices produced.
+    """
+    event_a = make_ethereum_event(1)
+    event_b = make_ethereum_event(2)
+    event_c = make_ethereum_event(3)
+
+    events = [event_a, event_b, event_c]
+    assert events == [event_a, event_b, event_c]  # sanity check
```
Comment (prettyirrelevant) [inline rotkehlchen/tests/unit/decoders/test_main.py:506]: You assigned the variable above yourself.

The `sequence_index` check is acceptable but this one seems like a tautology.

File: docs/changelog.rst [hunk 2]
```diff
@@ -9,6 +9,7 @@ Changelog
 * :bug:`-` Fixes bug where navigation bar on the left didn't expand by default.
 * :bug:`-` Fixes curve deposits and withdrawals accounting.
 * :bug:`5561` Fixes average cost basis calculation.
+* :bug:`-` Fixes bug when some decoded events in transactions would disappear.
```
Comment (LefterisJP) [inline docs/changelog.rst:12]:
```suggestion
* :bug:`-` Fixes a bug when some decoded events in transactions would disappear. 
```

**Review by LefterisJP — COMMENTED** [review]
It's in general okay, but I need to see a decoding test that fails when this change is not in. Basically the scenario you saw that lead you to this (perhaps as unit test also and simplified -- just need the scenario)

**Review by LefterisJP — APPROVED** [review]
LGTM. @nebolax explained that there were duplicated indices in `rotkehlchen/tests/unit/decoders/test_curve.py::test_deposit_multiple_tokens` which this change changes and fixes.

**Pull request merged**